### ISSUE TITLE
Traits-like Abilities Documentation (GNA #17432)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1245]
**[Original submission](https://gna.org/bugs/index.php?17432)** by gallaecio on 2011-01-07

I would like to ask for Abilities to be managed as traits in Wesnoth help. Currently, abilities male form is used as title, and ability description's first line is ignored (see screenshots).

Instead, as it's done with traits, it would be good to have male/female name (for ingame, as adjective), general name (to be used as title in the Help) and description, without ability "general name" inside it.

In short: abilities documentation managed the same way as trait's one (from a l10n point of view).

(Reproduced on any)
Release: 1.9.3
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by wesnoth-bugs** [comment]
**Modified on 2011-01-07**

**gallaecio added attachment:** [wesnoth.png](http://gna.org/bugs/download.php?file_id=11726)

**gallaecio added attachment:** [wesnoth2.png](http://gna.org/bugs/download.php?file_id=11727)

**Comment by CelticMinstrel** [comment]
Is this relevant anymore? The name appearing in the first line of the description is deprecated in 1.12 as I recall.

**Comment by Wedge009** [comment]
Yeah, I just checked, abilities and traits appear to be handled the same way in 1.14.

**Comment by jostephd** [comment]
This part of the request is not yet done:

> Instead, as it's done with traits, it would be good to have male/female name (for ingame, as adjective), general name (to be used as title in the Help) and description, without ability "general name" inside it.

Traits have name, male_name, and female_name. The help uses name in preference to male_name and female_name:

https://github.com/wesnoth/wesnoth/blob/de8a9133745a88840d02fb0a014f0afdd6624da3/src/help/help_impl.cpp#L684-L687

(That's for trait pages only. [Unit type pages](https://github.com/wesnoth/wesnoth/blob/de8a9133745a88840d02fb0a014f0afdd6624da3/src/help/help_topic_generators.cpp#L416-L424) work differently.)

Abilities don't support separate name and male_name:

https://wiki.wesnoth.org/AbilitiesWML#Common_keys_and_tags_for_every_ability

**Comment by jostephd** [comment]
> The help uses name in preference to male_name and female_name:

Sorry, it doesn't. But gallaecio implied that's how it should work, so maybe we ought to change that? :thinking: In any case, the request was to add separate name and male_name to abilities.
